packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -182,6 +183,20 @@ export interface UserPoolClientOptions {
    * @default true for new stacks
    */
   readonly preventUserExistenceErrors?: boolean;
+
+  /**
+   * Whether users registered in the user pool should be able to sign in using this client.
+   * If this is set to `true` and `identityProviders` are set, the client will only allow sign in via
+   * third-party identity providers.
+   * @default true
+   */
+  readonly allowUserPoolIdentities?: boolean;
+
+  /**
+   * The list of federated identity providers that users should be able to use to sign in using this client.
+   * @default - no identity providers
+   */
+  readonly identityProviders?: IUserPoolIdentityProvider[];
```

Comment:
Maybe `additionalIdentityProviders`? (In addition to Cognito itself?)

Reply:
`additional` feels awkward, Cognito uses the prefix `supported`; I'd prefer to leave this as is, unless you feel strongly.

OTOH if we go with the API suggested above, this property will be dropped.

Reply:
Either way we go this should probably be dropped, agreed.